libnd4j/include/ops/impl/specials_single.hpp

```diff
@@ -127,76 +127,130 @@ namespace sd {
+  for (int i = 0; i < numOfInArrs; ++i) {
+    shapeExtendedWithOnes = shapeExtendedWithOnes && isShapeExtendedWithOnes(*inArrs[i], axis);
+    followEws1 = followEws1 && inArrs[i]->ews() == 1;
+    matchesOutputOrdering = matchesOutputOrdering && inArrs[i]->ordering() == output.ordering();
   }
 
-  if (luckCase1) {  // for example {1,10} + {2,10} + {3,10} = {6, 10} order c; or {10,1} + {10,2} + {10,3} = {10, 6}
-                    // order f
-
-    T *z = zBuff;
-    for (sd::Unsigned i = 0; i < numOfInArrs; ++i) {
-      const auto memAmountToCopy = inArrs[i]->lengthOf();
-      memcpy(z, inArrs[i]->bufferAsT<T>(), memAmountToCopy * sizeofT);
-      z += memAmountToCopy;
+  bool copyCaseEws1 = followEws1 & matchesOutputOrdering;
+  bool copyCase1 = numOfInArrs > 1 ? copyCaseEws1 & shapeExtendedWithOnes : copyCaseEws1;
+  if (copyCase1) {
+    std::vector<T *> zPtrList;
+    T *z = output.bufferAsT<T>();
+    for (int i = 0; i < numOfInArrs; i++) {
+      zPtrList.push_back(z);
+      z += inArrs[i]->lengthOf();
     }
+    auto func = [&inArrs, &zPtrList](uint64_t thread_id, int64_t start, int64_t stop, int64_t increment) -> void {
+      for (int i = start; i < stop; ++i) {
+        const auto memAmountToCopy = inArrs[i]->lengthOf();
+        const auto inputPtr = inArrs[i]->bufferAsT<T>();
+#if defined(__NEC__)
+        auto zPtr = zPtrList[i];
+        for (int j = 0; j < memAmountToCopy; j++) {
+          zPtr[j] = inputPtr[j];
+        }
+#else
+        memcpy(zPtrList[i], inputPtr, memAmountToCopy * sizeof(T));
+#endif
+      }
+    };
+
+    samediff::Threads::parallel_tad(func, 0, numOfInArrs, 1);
     return;
   }
 
-  // const bool isZcontin = output.strideAt(axis) == 1;
-  // bool areInputsContin = true;
-  // bool allSameOrder    = true;
-  // std::vector<sd::LongType> strideOfContigStride(numOfInArrs);
-
-  // if(isZcontin) {
-
-  //     for (sd::Unsigned i = 0; i < numOfInArrs; ++i) {
-
-  //         areInputsContin &= inArrs[i]->strideAt(axis) == 1;
-  //         allSameOrder    &= inArrs[i]->ordering() == output.ordering();
-  //         if(!areInputsContin || !allSameOrder)
-  //             break;
-
-  //         strideOfContigStride[i] = strideOverContigAxis(axis, inArrs[i]->getShapeInfo());
-  //     }
-  // }
-
-  // const bool luckCase2 = isZcontin && areInputsContin && allSameOrder;
-
-  // if(luckCase2) {     // for example {2,1,3} + {2,5,3} + {2,10,3} = {2,16,3}, here axis 1 shoud have stride = 1 for
-  // all inputs arrays and output array
-
-  //     const auto zStep = strideOverContigAxis(axis, output.getShapeInfo());
-
-  //     for (sd::Unsigned i = 0; i < output.lengthOf() / output.sizeAt(axis); ++i) {
+  // for one Array
+  if (numOfInArrs < 2) {
+    output.assign(inArrs[0]);
+    return;
+  }
 
-  //         T* z = zBuff + zStep * i;
+  if (copyCaseEws1 && output.ordering() == 'c') {
+    int times = 1;
+    auto shapes = shape::shapeOf(output.shapeInfo());
+    assert(axis < rank);
```

Comment:
Add a message to this describing the problem. Random errors without messages are VERY jarring especially from java.

Reply:
I will remove it at all as it should be guaranteed already